### DISCUSSION TITLE
How do I use the built-in DAC in the ESP32?

### DISCUSSION THREAD

**Discussion by Blauenfeldt** [sle118/squeezelite-esp32#417] [Q&A]
I have a esp32-devkitC-v4 board, which use the ESP32-WROVER module, which uses the ESP32-D0WDQ6 chip.
The chip has 2 built-in DAC, at GPIO25 and GPIO26. Can I use those instead of an external DAC? It would build cheaper, smaller, and more simple. Then I would only need an amp. My thought was to use them for stereo or tweeter/subwoofer output.

[Board](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/_images/esp32-devkitC-v4-pinout.png)
[Module](https://cdn-shop.adafruit.com/product-files/3384/esp32-wrover_datasheet_en.pdf)
[Chip](https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf)

After posting: I see the DAC is 8 bit. Isn't that gonna make the sound, sound bad?

**Comment by Renber78** [comment]
Hi, I don't think internal DAC support is included in this project.
But as you announced, it only has 8-bit... it is obvious that the sound will not be Hifi... plus you will get parasitic noise from the ESP's internal power supply. Not really great!
For 2-3€ you can buy an external DAC like a PCM5102 which gives good results.
